base/bucket_gocb.go

```diff
@@ -1053,29 +1092,96 @@ func (bucket CouchbaseBucketGoCB) GetWithXattr(k string, xattrKey string, rv int
 
 }
 
-// Delete a document and it's associated named xattr.  Couchbase server will preserve system xattrs as part of the (CBS) tombstone when a document is deleted.
-// To remove the system xattr as well, an explicit subdoc delete operation is required.
+// Only called for Purge
 func (bucket CouchbaseBucketGoCB) DeleteWithXattr(k string, xattrKey string) error {
 
+	return bucket.deleteWithXattrInternal(k, xattrKey, nil)
```

Comment:
It looks like we don't have any retry handling in DeleteWithXattr for transient server errors.  That might be appropriate - since it's not part of replication, it would allow callers to do their own throttling when hitting load-related errors.